server/service/src/item/item_variant/test/mod.rs

```diff
@@ -0,0 +1,154 @@
+                UpsertItemVariant {
+                    id: uuid(),
+                    item_id: mock_item_b().id,
+                    name: "item_b_variant_a".to_string(),
+                    ..Default::default()
+                },
+            )
+            .unwrap();
+
+        // Query the item variant by name
+        let item_variant = service
+            .get_item_variants(
+                &context,
+                None,
+                Some(ItemVariantFilter::new().name(StringFilter::equal_to("item_a_variant_a"))),
+                None,
+            )
+            .unwrap();
+
+        assert_eq!(item_variant.count, 1);
+        assert_eq!(item_variant.rows[0].id, test_item_a_variant_id);
+
+        // Update the name
+        let _item_variant = service
+            .upsert_item_variant(
+                &context,
+                UpsertItemVariant {
+                    id: test_item_a_variant_id.to_string(),
+                    item_id: mock_item_a().id,
+                    name: "updated_name".to_string(),
+                    ..Default::default()
+                },
+            )
+            .unwrap();
+
+        // Query the item variant by name
+        let item_variant = service
+            .get_item_variants(
+                &context,
+                None,
+                Some(ItemVariantFilter::new().name(StringFilter::equal_to("updated_name"))),
+                None,
+            )
+            .unwrap();
+
+        assert_eq!(item_variant.count, 1);
+        assert_eq!(item_variant.rows[0].id, test_item_a_variant_id);
+
+        // Query the item variant by id
+        let item_variant = service
+            .get_item_variants(
+                &context,
+                None,
+                Some(ItemVariantFilter::new().id(EqualFilter::equal_to(test_item_a_variant_id))),
+                None,
+            )
+            .unwrap();
+
+        assert_eq!(item_variant.count, 1);
+        assert_eq!(item_variant.rows[0].id, test_item_a_variant_id.to_string(),);
+
+        // Delete the item variant
+        service
+            .delete_item_variant(
+                &context,
+                DeleteItemVariant {
+                    id: test_item_a_variant_id.to_string(),
+                },
+            )
+            .unwrap();
+
+        // Check that the delete worked
+        let item_variant = service
+            .get_item_variants(
+                &context,
+                None,
+                Some(ItemVariantFilter::new().id(EqualFilter::equal_to(test_item_a_variant_id))),
+                None,
+            )
+            .unwrap();
+
+        assert_eq!(item_variant.count, 0);
+    }
+
+    #[actix_rt::test]
+    async fn validate_item_variant() {
+        // TODO validation tests
+
+        // Test that the item variant name is set?
+
+        // Test that we can't create a record with an item_id that doesn't exist
+
+        // Test that we can't change the item_id on an existing record???
```

Comment:
Hm, this one is a good point! I guess fine to determine if we are doing update or insert in the validation and apply some different checks accordingly! Same with the unique name check